server/start.go

```diff
@@ -220,8 +222,18 @@ func startInProcess(ctx *Context, cdc codec.JSONMarshaler, appCreator AppCreator
 		apiSrv = api.New(ctx)
 		app.RegisterAPIRoutes(apiSrv)
 
-		if err := apiSrv.Start(config); err != nil {
+		errCh := make(chan error)
+
+		go func() {
+			if err := apiSrv.Start(config); err != nil {
+				errCh <- err
+			}
+		}()
+
+		select {
+		case err := <-errCh:
 			return err
+		case <-time.After(5 * time.Second): // assume server started successfully
```

Comment:
I did not realize TM HTTP client actually blocks -- this fixes that so we can properly cleanup below.